api/response/error.go

```diff
@@ -61,6 +62,19 @@ func NewError(code int, err string) *ErrorResp {
 	}
 }
 
+func NewInternalError(err string) *ErrorResp {
```

Comment:
It doesnt sit well with me to have idx/memory depend on api/response when the idx has a lot of uses that use the api.

response.WrapError() will set the status code correctly if the pass in error implements the response.Error interface.

Reply:
I agree. ideally none of the internal libraries should import any api libraries. to be more precise: none of the internal libraries should be concerned about http implementation details.
but those internal libraries should be able to denote whether an error is caused by user, or caused by the MT platform. the most practical way to do this is still use the http codes (hence your Error interface).
we could create an internal error type that can denote the difference between these two scenarios (and perhaps a few more) but invariably it would tie into http status codes anyway, so I figured, might as well put it with the rest of that stuff.

what's the alternative? create another internal error type outside of `api/response`, but it would still have to be able to set code 500, 400, etc (so practically it would still import the http package because it allows us to refer to `http.StatusBadRequest` and so forth). Would that be your preferred approach? (it would be mine but i abandoned that approach and i forgot why)

Reply:
> create another internal error type outside of api/response

That would be my approach.  It seems excessive to have to re-define an error struct that impliments the response.Error interface in every package. So maybe we just create a top level
metrictank/errors package and move instance.Error and instance.ErrorResp there (or just copy them for now and open another issue to update all of the api package to use it)